unidist/core/backends/mpi/core/controller/common.py

```diff
@@ -144,6 +145,7 @@ def request_worker_data(data_id):
         # without any delay
         "is_blocking_op": True,
     }
+    # a blocking send is used because completion is necessary for processing to continue
```

Comment:
```suggestion
    # We use a blocking send here because we have to wait for
    # completion of the communication, which is necessary for the pipeline to continue.
```